lib/nerves_hub_link_common/fwup_config.ex

```diff
@@ -3,10 +3,13 @@ defmodule NervesHubLinkCommon.FwupConfig do
   Config structure responsible for handling callbacks from FWUP,
   applying a fwupdate, and storing fwup task configuration
   """
+  alias NervesHubLinkCommon.UpdateAvailable
+
   defstruct fwup_public_keys: [],
             fwup_devpath: "/dev/mmcblk0",
             handle_fwup_message: nil,
-            update_available: nil
+            update_available: nil,
+            journal_location: "/tmp/"
```

Comment:
```suggestion
            journal_location: "/data/nerves_hub_link.partial"
```

I'm just making up the journal location so other names work for me too. What I'd like to suggest are:

1. Putting it on a persistent filesystem so that it can survive a reboot
2. Only having one file to make it easy to figure out where to find it and easy to clean up

I haven't gotten to it yet, but I'm guessing that the use of one file might have other ramifications. I do like the simplicity and limiting the number of files that it can create.

Reply:
yeah i only defaulted it to /tmp for the test suite since `/data` can't be written to on our host machines. I agree on keeping a single file, but i also wanted this to be a directory because there needs to be some way of identifying a partial download as a particular firmware after a reboot, which i was going to use the Filename for. 
I also considered putting a header in the partial file, but am still thinking about that.